File: perl/ase_input_conf_1.pl
#!/usr/bin/perl
use strict;

# Automation Scripting Essential Skills 
# Perl

# Input: Reading configuration files
# Technique #1 -- Use angle brackets and a regex

our $Setting;
our $Value;

if( $#ARGV >= 1 )
{
    $Setting = $ARGV[1];
    # pop the second arg off so that the 
    # while loop below doesn't try to open
    # a file named after the setting.
    $ARGV = pop @ARGV;
}

while (<>)
{
    if( /^([^=]+)\s*=\s*(.*)$/ )
    { 
        my $thisSetting = $1;
        my $thisValue = $2;
        if( $thisSetting =~ /^$Setting$/ ) 
        {
            print "thisSetting = $thisValue\n";
        }
        else
        {}
    }
    else
    {
        print "no match\n";
    }
}

File: python/ase_input_conf_1.py
#!/usr/bin/env python


# Automation Scripting Essential Skills 
# Python

# Input: Reading configuration files
# Technique1 -- Use ConfigParser and an ini-type file 
import sys
import ConfigParser

# SafeConfigParser supports interpolation
config = ConfigParser.SafeConfigParser()
config.read(sys.argv[1])

print config.get('root', sys.argv[2])
